.ci/CI/src/modules/IntegTest/Graph.jl

```diff
@@ -0,0 +1,239 @@
+"""
+    _git_clone(repo_url::AbstractString, directory::AbstractString)
+
+Clones git repository
+
+# Args
+- `repo_url::AbstractString`: Url of the repository. Can either be a plan URL or use the Julia Pkg
+    notation <URL>#<branchname>. e.g. https://https://github.com/user/repo.git#dev to clone the
+    dev branch of the repository `repo`.
+- `directory::AbstractString`: Path where the cloned repository is stored.
+"""
+function _git_clone(repo_url::AbstractString, directory::AbstractString)
+    splitted_url = split(repo_url, "#")
+    return if length(splitted_url) < 2
+        _git_clone(repo_url, "dev", directory)
+    else
+        _git_clone(splitted_url[1], splitted_url[2], directory)
+    end
+end
+
+"""
+    _git_clone(repo_url::AbstractString, directory::AbstractString)
+
+Clones git repository
+
+# Args
+- `repo_url::AbstractString`: Url of the repository.
+- `branch::AbstractString`: Git branch name
+- `directory::AbstractString`: Path where the cloned repository is stored.
+"""
+function _git_clone(
+        repo_url::AbstractString, branch::AbstractString, directory::AbstractString
+    )
+    @info "clone repository: $(repo_url)#$(branch) -> $(directory)"
+    return with_logger(debuglogger) do
+        try
+            @debug "git clone --depth 1 -b $(branch) $(repo_url) $directory"
+            run(
+                pipeline(
+                    `git clone --depth 1 -b $(branch) $(repo_url) $directory`;
+                    stdout = devnull,
+                    stderr = devnull,
+                ),
+            )
+        catch
+            @debug "LibGit2.clone($(repo_url), $(directory); branch=$(branch))"
+            LibGit2.clone(repo_url, directory; branch = branch)
+        end
```

Comment:
What is being caught here? It looks like the git clone is expected to fail sometimes?

Reply:
`LibGit2` does not support the git clone depth. Therefore some clones are pretty slow. So I use the `git` executable, if available. The code actual tries to call `git clone --depth ...`. If the executable is available, everything is fine. If not, an error is thrown and the fallback is used.

I cannot exactly remember, why I solved it in this way, but I think there is no Julia function to check, if an executable is available.